packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts

```diff
@@ -286,7 +286,7 @@ export function buildBreadcrumbManager(
   }
```

Comment:
Don't let yourself be fooled by Svelte-Kit new-ness :smile: 

The issue actually happens with vanilla Vite, so I'm currently investigating whether it actually comes from Rollup

Reply:
Could not reproduce in rollup, so I created this: https://github.com/vitejs/vite/issues/3247

Point remains that this is still something that breaks really hard and it is painful to debug :sweat: 

I really don't mind rejecting this, I mostly wanted to discuss the issue. If you don't mind this breaking in some cases, so be it

---

> I also see headless-search-box is adjusted, but this controller is composed inside headless-standalone-search-box

I searched for `...controller`, didn't notice there were other variable names for this one

Reply:
> Point remains that this is still something that breaks really hard and it is painful to debug

I agree, but I think we should push to have the issue resolved inside vite rather than adjust headless. Some benefits of this:
1. The fix will be available to all projects using vite, helping the wider js community.
2. Developers working inside headless can continue to use standard ES6 spread. The code-base stays free of unnatural patterns that are hard for us to enforce and likely to regress in the future.

I think we can be patient in this case since an external client is not asking for it, and the issue is caused by a specific 3rd-party package, not headless which is using standard syntax.

Reply:
Seconding what Sami said:

If we absolutely need to have a quirk for a particular toolchain: so be it, there comes a point where we can't really do anything. A good example is Salesforce Locker Service, which forces us to have some quirks in how we do things in certain places.

When that happens, however, it means we need to have a great "explanation" + try to isolate the quirk as much as possible. 

In Locker Service case, it's a business requirement (Salesforce is obviously a very important ecosystem for the company+ it's understandable/good reasons why some things in Headless will *never* work in Locker Service , so there's nothing to report/fix on Salesforce's side).

In this scenario, if we want to have something really future proof, it would means refactoring how we build controller in the first place (in base controller, or with some utility function) + to possibly have some kind of test to make sure we aren't breaking this. Otherwise, there's going to be regression, pretty much 100% assured.

So... a much more involved project, unfortunately.